resource-manager/src/main/java/org/snomed/module/storage/ModuleStorageCoordinator.java

```diff
@@ -0,0 +1,602 @@
+            if (resourceManager.doDoesObjectExist(metadataResourcePath)) {
+                ModuleMetadata moduleMetadata = FileUtils.convertToObject(resourceManager.doReadResourceFile(metadataResourcePath), ModuleMetadata.class);
+                if (moduleMetadata != null) {
+                    String rf2ResourcePatch = baseResourcePath + "/" + moduleMetadata.getFilename();
+                    if (resourceManager.doDoesObjectExist(rf2ResourcePatch)) {
+                        if (includeFile) {
+                            moduleMetadata.setFile(resourceManager.doReadResourceFile(rf2ResourcePatch));
+                        }
+
+                        return moduleMetadata;
+                    }
+                }
+            }
+        }
+
+        return null;
+    }
+
+    private boolean doArchiveRelease(String codeSystem, String moduleId, String effectiveTime) {
+        LOGGER.trace("Archiving {}/{}_{}/{}", writeDirectory, codeSystem, moduleId, effectiveTime);
+
+        String metadataResourcePath = getMetadataResourcePath(writeDirectory, codeSystem, moduleId, effectiveTime);
+        if (!resourceManager.doDoesObjectExist(metadataResourcePath)) {
+            return false;
+        }
+
+        ModuleMetadata moduleMetadata = FileUtils.convertToObject(resourceManager.doReadResourceFile(metadataResourcePath), ModuleMetadata.class);
+        if (moduleMetadata == null) { // De-serialisation failed
+            return false;
+        }
+
+        String packageResourcePath = getPackageResourcePath(writeDirectory, codeSystem, moduleId, effectiveTime, moduleMetadata.getFilename());
+        if (!resourceManager.doDoesObjectExist(packageResourcePath)) {
+            return false;
+        }
+
+        String epochSecond = Long.toString(Instant.now().getEpochSecond());
+        String metadataArchivePath = asArchivePath(metadataResourcePath, epochSecond);
+        String packageArchivePath = asArchivePath(packageResourcePath, epochSecond);
+        boolean metadataCopied = resourceManager.doCopyResource(metadataResourcePath, metadataArchivePath);
+        if (!metadataCopied) {
+            return false;
+        }
+
+        boolean packageCopied = resourceManager.doCopyResource(packageResourcePath, packageArchivePath);
+        if (!packageCopied) {
+            return false;
+        }
+
+        boolean metadataDeleted = resourceManager.doDeleteResource(metadataResourcePath);
+        if (!metadataDeleted) {
+            return false;
+        }
+
+        boolean packageDeleted = resourceManager.doDeleteResource(packageResourcePath);
+        return packageDeleted;
+    }
+
+    private String asArchivePath(String resourcePath, String epochSecond) {
+        // otf-common/files/ABC_12345/20240101/metadata.json => // otf-common/files/ABC_12345/archive/$epochSecond/metadata.json
+        return resourcePath.replaceAll("/\\d{8}/", "/archive/" + epochSecond + "/");
+    }
```

Comment:
@pgwilliams This shows how metadata.json moves from its directory to the archive directory.